src/hud.cpp

```diff
@@ -87,24 +87,31 @@ Hud::Hud(video::IVideoDriver *driver, scene::ISceneManager* smgr,
 	m_halo_boxes.clear();
 
 	m_selection_pos = v3f(0.0, 0.0, 0.0);
-	std::string mode = g_settings->get("node_highlighting");
+	std::string mode_setting = g_settings->get("node_highlighting");
+
+	if (mode_setting == "halo") {
+		m_mode = HIGHLIGHT_HALO;
+	} else if (mode_setting == "box") {
+		m_mode = HIGHLIGHT_BOX;
+	} else {
+		m_mode = HIGHLIGHT_NONE;
```

Comment:
The default for invalied values should be `HIGHLIGHT_BOX`, as it's the common and most used outlining style.